packages/devtools_app/lib/src/timeline/flutter/flutter_frames_chart.dart

```diff
@@ -2,20 +2,669 @@
+    // with a NAN.
+    _frameDurations.add(createBarEntry(0));
+
+    if (!simulateFeed) {
+      _loadAllData(3);
+    } else if (_timer == null) {
+      _startFeed(3);
+    } else {
+      return;
+    }
+
+    // Create heap used dataset.
+    frameDurationsSet = BarDataSet(_frameDurations, 'UI')
+      ..setColors1([mainGpuColorLight, mainUiColorLight])
+      ..setDrawValues(false);
+
+    // Create a data object with all the data sets - stacked bar.
+    _chartController.data = BarData([]..add(frameDurationsSet));
+
+    // specify the width each bar should have
+    _chartController.data.barWidth = 0.8;
+  }
+
+  // TODO(terry): Consider grouped bars (UI/GPU) not stacked.
+  BarEntry createBarEntry(int index) {
+    assert(index < _cannedData.length);
+    final x = _cannedData[index].toDouble();
+    final uiDurationValue = _cannedData[index + 1].toDouble();
+    final gpuDurationValue = _cannedData[index + 2].toDouble();
+
+    // TODO(terry): Structured item 0 is UI vs item 1 is GPU if not stacked.
+    return BarEntry.fromListYVals(
+      x: x,
+      vals: [
+        gpuDurationValue,
+        uiDurationValue,
+      ],
+    );
+  }
+
+  // Simulate live feed.
+  void _startFeed(int startIndex) {
+    // Fetch from the beginning of the canned data for the live feed.
+    timerDataIndex = startIndex; // Entry zero is already primed.
+
+    // TODO(terry): Consider moving the timer to the MemoryController.
+    // TODO(terry): The chart should be notified when new data arrives from the controller
+    //              using a notifier pattern.
+    // Average VMSerice rate is ~500-600 ms?
+    _timer = Timer.periodic(const Duration(milliseconds: 60), (Timer timer) {
+      if (timerDataIndex == 0) {
+        // First time reset our plotted data.
+        setState(() {
+          _frameDurations.clear();
+        });
+      }
+
+      // Pause pressed stop pumping out data simulating a live feed.
+      if (timerDataIndex < _cannedData.length) {
+        _frameDurations.add(createBarEntry(timerDataIndex));
+        timerDataIndex += 3;
+      }
+
+      updateChart();
+    });
+  }
+
+  void updateChart() {
+    _chartController.data = BarData([]..add(frameDurationsSet));
+
+    setState(() {
+      // Signal data has changed.
+      frameDurationsSet.notifyDataSetChanged();
+    });
+  }
+
+  void _loadAllData(int startIndex) {
+    int index = startIndex;
+    while (index < _cannedData.length) {
+      _frameDurations.add(createBarEntry(index));
+      index += 3;
+    }
+  }
+
   @override
   Widget build(BuildContext context) {
-    return Padding(
-      padding: const EdgeInsets.only(bottom: 8.0),
-      child: Container(
-        height: 150.0,
-        color: Colors.black26,
-        child: const Center(
-          child: Text('TODO Flutter Frames Chart'),
-        ),
+    return Container(
+      height: 200.0,
+      child: BarChart(_chartController),
+    );
+  }
+}
+
+class YAxisMilliSecFormatter extends ValueFormatter {
```

Comment:
nit: maybe rename to YAxisMillisFormatter or YAxisMillisecondFormatter or YAxisUnitFormatter, etc.

Reply:
Done YAxisUnitFormatter